uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -920,28 +920,31 @@ public static function findOverlappingLeaveRequests($leaveRequestParams, $leaveR
       ? "lr.from_date < '{$toDateTime}' AND lr.to_date > '{$fromDateTime}'"
       : "lrd.date BETWEEN '{$fromDate}' AND '{$toDate}'";
 
-    if ($isCalculationUnitInHours) {
-      $absenceTypesSelector = 'AND lr.type_id IN (' . implode(',', $absenceTypesIDsWithSameUnit) . ')';
-
-      if (count($absenceTypesIDsWithDifferentUnit)) {
-        $considerRequestsInDifferentUnit =
-          "OR ((lrd.date BETWEEN '{$fromDate}' AND '{$toDate}') AND lr.type_id IN (" .
-          implode(',', $absenceTypesIDsWithDifferentUnit) . "))";
+    if (!$isTOIL) {
+      if ($isCalculationUnitInHours) {
+        $absenceTypesSelector = 'AND lr.type_id IN (' . implode(',', $absenceTypesIDsWithSameUnit) . ')';
+
+        if (count($absenceTypesIDsWithDifferentUnit)) {
+          $considerRequestsInDifferentUnit =
+            "OR ((lrd.date BETWEEN '{$fromDate}' AND '{$toDate}') AND lr.type_id IN (" .
+            implode(',', $absenceTypesIDsWithDifferentUnit) . "))";
+        }
+      } else {
+        $absenceTypesSelector = 'AND lr.type_id IN (' . implode(',', $absenceTypesIDs) . ')';
       }
     }
 
-    if (!$isCalculationUnitInHours && !$isTOIL) {
-      $absenceTypesSelector = 'AND lr.type_id IN (' . implode(',', $absenceTypesIDs) . ')';
-    }
-
+    $ignoreRequestsWithDifferentUnit = count($absenceTypesIDsWithDifferentUnit)
+      ? 'OR lr.type_id IN (' . implode(',', $absenceTypesIDsWithDifferentUnit) . ')'
```

Comment:
Shouldn't it start with an 'AND'?

Reply:
Also is it actually ignoring requests with different units or taking them into consideration?

Reply:
Should be OR . Changed to "consider" - I think that's what was confising.